modules/nw-how-nw-iface-selected.adoc

```diff
@@ -20,8 +20,9 @@ As a result, the kubelet service uses the selected IP address as the node IP add
 If hardware or networking is reconfigured after installation, it is possible that the `nodeip-configuration.service` service can select a different NIC after a reboot.
 In some cases, you might be able to detect that a different NIC is selected by reviewing the `INTERNAL-IP` column in the output from the `oc get nodes -o wide` command.
 
-If network communication is disrupted or misconfigured because a different NIC is selected, one strategy for overriding the selection process is to set the correct IP address explicitly.
-The following list identifies the high-level steps and considerations:
+If network communication is disrupted or misconfigured because a different NIC is selected, you can create a hint file that includes the `NODEIP_HINT` variable to override the default IP selection logic. To create a hint file, see Optional: Overriding the default node IP selection logic. 
+
+In some situations, creating a hint file might be insufficient for overriding the selection process. The following list identifies the high-level steps and considerations to manually override the IP address: 
 
 * Create a shell script that determines the IP address to use for {product-title} communication. Have the script create a custom unit file such as `/etc/systemd/system/kubelet.service.d/98-nodenet-override.conf`. Use the custom unit file, `98-nodenet-override.conf`, to set the `KUBELET_NODE_IP` environment variable to the IP address.
 
```

Comment:
I'm wondering if this ever worked with OVNK. If you overrode the kubelet node ip then configure-ovs was still going to select its interface based on its own logic. The main point of this feature was to ensure that kubelet and configure-ovs had consistent ip/interface selection. Maybe it's worth noting that this old procedure is an OpenShift SDN-only thing.

We could expose the /run/nodeip-configuration stuff, but I'm hesitant to do that unless it turns out the NODEIP_HINT is insufficient for some use cases. I look at that as more of an internal interface, where NODEIP_HINT is the user-facing interface.

Reply:
It kinda worked for vSphere UPI no apiVIP multi-NIC  https://bugzilla.redhat.com/show_bug.cgi?id=2057160#c40

```
vSphere UPI with KUBELET_NODE_IP set and mixed static and DHCP

# /etc/systemd/system/kubelet.service.d/98-nodenet-override.conf
[Service]
Environment="KUBELET_NODE_IP=172.31.248.76" "KUBELET_NODE_IPS=172.31.248.76"
```

QE tests bonding with vSphere UPI `platform: vsphere` with no apiVIP where `nodeip-configuration` doesn't run, so we would still need a work around.

Reply:
err, well we test vSphere UPI `platform: vsphere` with no apiVIP for multi-NIC scenarios in general, bonding or not.

Reply:
I assume we don't have a choice. If node-ip is not running, we need something that will fill env variables and primary-ip file for ovs.
Though i wonder why node-ip is not running in that case? How it is different from None platform?

Reply:
See also https://issues.redhat.com/browse/OCPBUGS-2203

Reply:
VSphere UPI is unique because it doesn't use the None platform type. It uses VSphere, but it doesn't set the VIP fields. I think not enabling nodeip-configuration there was probably an accidental oversight, but we're discussing how to deal with it right now elsewhere.

We can't assume nodeip-configuration is always running anyway. It doesn't run on the cloud IPI platforms either so configure-ovs will always have to have its own selection logic. However, in the cloud there's usually only one interface anyway so it doesn't matter as much.